### PR TITLE
Measure upload speed

### LINKED ISSUE #27: Support fast.com's new upload speed test

### PR DESCRIPTION
**Pull request by GloriousYellow**
Closes #27

### CODE REVIEW

**Comment by GloriousYellow** [comment]
Upload speed measuring works, however a flag is not yet implemented

**Comment by GloriousYellow** [comment]
Progress display still feels a bit ugly though
image
image
image

**Comment by mattstrayer** [comment]
Can this get merged in?

File: cli.js [hunk 1]
```diff
@@ -7,10 +7,11 @@ const logUpdate = require('log-update');
 const ora = require('ora');
 const api = require('./api');
 
-meow(`
+const cli = meow(`
 	Usage
 	  $ fast
-	  $ fast > file
+		$ fast > file
+		$ fast --verbose
```
[the remark this answers is not in the document]
Comment (sindresorhus) [reply cli.js]: You also need to update the readme with the new CLI help output.

File: cli.js [hunk 2]
```diff
@@ -24,13 +25,30 @@ dns.lookup('fast.com', err => {
...
 
-const speed = () => chalk[data.isDone ? 'green' : 'cyan'](data.speed + ' ' + chalk.dim(data.unit)) + '\n\n';
+const speed = () =>
+	chalk[data.isDone ? 'green' : 'cyan'](
+		data.downloadSpeed +
+			' ' +
+			chalk.dim(data.downloadUnit) +
+			(cli.flags.verbose ? (
+				chalk.gray(' / ') +
+				(data.uploadSpeed || '-') +
```
Comment (sindresorhus) [inline cli.js]: I think the `-` should be `chalk.dim`, so the whole `- Mbps` thing is dim.

**Comment by sindresorhus** [comment]
Maybe we should use the `↓` `↑` arrows to indicate download/upload?

**Comment by sindresorhus** [comment]
> Can this get merged in?

It will be merged when it's ready. If you want it merged faster, help out by reviewing and testing this PR ;)

**Comment by sindresorhus** [comment]
@GloriousYellow This is looking great! 🙌

**Comment by HaShAkO** [comment]
Hi, is there any plan to include latency as well ? , great job =)

**Comment by havish10** [comment]
prevResult is not being set to anything in api.js

**Comment by GloriousYellow** [comment]
I completely forgot about this :/
I'll implement the changes this week

**Comment by GloriousYellow** [comment]
I'm thinking of using ⇣⇡ arrows, like in [pure](https://github.com/sindresorhus/pure). How does that sound?

**Comment by sindresorhus** [comment]
@GloriousYellow The last one looks good. Let's go with that.

**Comment by WilhelmErasmus** [comment]
Any updates on this?

**Comment by GloriousYellow** [comment]
I am not completely satisfied with the naming of `--verbose`; it doesn't imply measurement of upload speed.

**Comment by GloriousYellow** [comment]
image
Should we make the download speed indicator green during the upload test?

**Comment by GloriousYellow** [comment]
image

**Comment by GloriousYellow** [comment]
image
When completed

**Comment by mattstrayer** [comment]
I think it looks better with just the blue. The animation on the left indicates enough to the user when a test is running and when it is complete

**Comment by sindresorhus** [comment]
> Should we make the download speed indicator green during the upload test?

👍

**Comment by sindresorhus** [comment]
Thank you for your work on this, @GloriousYellow :)

**Comment by GloriousYellow** [comment]
There is a failing test, would that be a concern?

**Comment by sindresorhus** [comment]
Already fixed in a follow-up commit.

**Pull request merged**